### PR TITLE
Created problem_63 in project_euler

### PR DESCRIPTION
**Pull request by Kush1101**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Kush1101** [comment]
@cclauss Please review it.

File: project_euler/problem_63/sol1.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+The maximum base can be 9 because all n-digit numbers < 10^n.
+Now 9**23 has 22 digits so the maximum power can be 22.
+Using these conclusions, we will calculate the result.
+"""
+
+
+def compute_nums(max_base: int = 10, max_power: int = 22) -> int:
+    """
+    Returns the count of all n-digit numbers which are nth power
+    >>> compute_nums(10,22)
```
Comment (cclauss) [inline project_euler/problem_63/sol1.py:17]:
```suggestion
    >>> compute_nums(10, 22)

```

File: project_euler/problem_63/sol1.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    49
+    """
+    bases = list(range(1, max_base))
+    powers = list(range(1, max_power))
+    count = 0
+    for base in bases:
+        for power in powers:
+            if len(str((base ** power))) == power:
+                count += 1
+    return count
```
Comment (cclauss) [inline project_euler/problem_63/sol1.py:27]:
```suggestion
    return sum([1 for power in range(1, max_power) for base in range(1, max_base) if len(str(base ** power)) == power])
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: project_euler/problem_63/sol1.py (hunk 2 above)
Comment (Kush1101) [reply project_euler/problem_63/sol1.py:28]: I honestly thought you would say this. 😁

File: project_euler/problem_63/sol1.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+"""
+
+
+def compute_nums(max_base: int = 10, max_power: int = 22) -> int:
+    """
+    Returns the count of all n-digit numbers which are nth power
+    >>> compute_nums(10, 22)
+    49
+    """
+    bases = list(range(1, max_base))
```
Comment (cclauss) [inline project_euler/problem_63/sol1.py:20]: If max_base and/or max_power is large then the list() call could occupy a lot of RAM and be slower than using the range iterator without the list() call.

Comment (cclauss) [reply project_euler/problem_63/sol1.py:20]:
```
import sys
>>> sys.getsizeof(range(1000000))
48
>>> sys.getsizeof(list(range(1000000)))
9000112
```

Comment (Kush1101) [reply project_euler/problem_63/sol1.py:20]: Should I consider removing the parameters?

Comment (Kush1101) [reply project_euler/problem_63/sol1.py:20]: Ok. I got it.

**Comment by cclauss** [comment]
Please add doctests for 1, 0, -1

**Review by cclauss — APPROVED** [review]

**Pull request merged**